src/storage/page_buffer.c

```diff
@@ -7713,6 +7714,7 @@ pgbuf_claim_bcb_for_fix (THREAD_ENTRY * thread_p, const VPID * vpid, PAGE_FETCH_
 #endif /* ENABLE_SYSTEMTAP */
 
       pgbuf_request_data_page_from_page_server (vpid);
+      // TODO: Ilie Wait here for the page?
```

Comment:
The wait should happen after reading the page from disk. Better to remove TODO comment for now.